src/components/vote-form.js

```diff
@@ -1,14 +1,121 @@
-import React from "react";
-import styled from "styled-components";
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
 
-export default function VoteForm() {
-  return <Wrapper>안녕 나는 투표 폼!</Wrapper>;
+import styled from 'styled-components';
+
+function VoteForm() {
+	const [candi, setCandi] = useState([]);
+	const [voteCount, setVoteCount] = useState();
+
+	useEffect(() => {
+		getCandidates();
+	}, []);
+
+	const getCandidates = async () => {
+		await axios
+			.get(process.env.API_HOST + '/candidates/', candi)
+			.then(({ data }) => {
+				setCandi(data);
+			})
+			.catch(function (error) {
+				console.log(error);
+			});
+	};
+
+	const voteCandidates = async (person) => {
+		const newUrl =
+			process.env.API_HOST + '/candidates/' + person._id + '/vote/';
+		await axios
+			.put(newUrl, voteCount)
+			.then(({ data }) => {
+				setVoteCount(data);
+				alert(person.name + '님에게 투표 완료!');
+				getCandidates();
+			})
+			.catch(function (error) {
+				console.log(error);
+				alert('투표 실패!');
+			});
+	};
+	let i = 1;
+	return (
+		<Wrapper>
+			<Title>
+				<RedTitle>프론트엔드 인기쟁이</RedTitle>는 누구?
+			</Title>
+			<SubTitle>CEOS 프론트엔드 개발자 인기 순위 및 투표 창입니다.</SubTitle>
+			<VoteArea>
+				{candi
+					.sort((person1, person2) => {
+						return person2.voteCount - person1.voteCount;
+					})
+					.map((person) => (
+						<Row key={JSON.stringify(person)}>
+							<Rank>{i++}위:</Rank>
+							<CandiName>
+								{person.name}
+								<br />[{person.voteCount}표]
+							</CandiName>
+							<Button onClick={() => voteCandidates(person)}>투표</Button>
+						</Row>
+					))}
+			</VoteArea>
+		</Wrapper>
+	);
 }
+export default React.memo(
+	VoteForm,
+	(prev, next) => prev.voteCount === next.voteCount
+);
```

Comment:
VoteForm component에는 prop이 없죠! React.memo는 prop을 비교해서 리렌더를 결정하기 때문에, prop이 없는 component는 아래처럼 비교함수를 안 써주시면 됩니다.
```suggestion
export default React.memo(VoteForm);
```